packages/effective-module-tree/index.js

```diff
@@ -0,0 +1,109 @@
+const fs = require( 'fs' ).promises;
+const path = require( 'path' );
+const globby = require( 'globby' );
+const debug = require( 'debug' )( 'effective-module-tree' );
+const treeify = require( 'object-treeify' );
+
+// Generator that yields all places were node will look for modules
+// For a path like : a/node_modules/b/node_modules/c
+// It will generate the candidates:
+//		- a/node_modules/b/node_modules/c/node_modules
+// 		- a/node_modules/b/node_modules
+//		- a/node_modules
+//		- node_modules
+const candidates = function*( packagePath ) {
+	const parts = packagePath.split( path.sep );
+	for ( let i = parts.length; i >= 0; i-- ) {
+		// This avoid generating .../node_modules/node_modules/...
+		if ( parts[ i - 1 ] === 'node_modules' ) continue;
```

Comment:
style nit: we usually do:
```js
if ( x ) {
  continue;
}
```

The formatting choices start to matter when one wants to step through the code with debugger and set breakpoints. Having a critical statement on a separate line helps a lot.

Certain kinds of code need to be basically rewritten before I can start debugging. 🙂

Reply:
Can we encode that an eslint rule?

Reply:
> Can we encode that as an eslint rule?

Seems like [`curly`](https://eslint.org/docs/rules/curly) would cover that. I'm surprised we don't have the rule enabled.